src/overlays/actors/ovl_Obj_Syokudai/z_obj_syokudai.c

```diff
@@ -27,11 +29,243 @@ const ActorInit Obj_Syokudai_InitVars = {
+    this->lightNode = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, &this->lightInfo);
+    
+    if ((this->actor.params & 0x400) || ((torchType != 2) && Flags_GetSwitch(globalCtx, this->actor.params & 0x3F))) {
+        this->timer = -1;
+    }
+    
+    this->flameTexScroll = (s32)(Math_Rand_ZeroOne() * 20.0f);
+    sLitTorchCount = 0;
+    Actor_SetHeight(&this->actor, 60.0f);
+}
+
+void ObjSyokudai_Destroy(Actor* thisx, GlobalContext *globalCtx) {
+    s32 pad;
+    ObjSyokudai* this = THIS;
+    
+    Collider_DestroyCylinder(globalCtx, &this->collider1);
+    Collider_DestroyCylinder(globalCtx, &this->collider2);
+    LightContext_RemoveLight(globalCtx, &globalCtx->lightCtx, this->lightNode);
+}
+
+void ObjSyokudai_Update(Actor* thisx, GlobalContext *globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    ObjSyokudai* this = THIS;
+    s32 torchCount = (this->actor.params >> 6) & 0xF;
+    s32 switchFlag = this->actor.params & 0x3F;
+    s32 torchType = this->actor.params & 0xF000;
+    s32 litTimeScale;
+    WaterBox* dummy;
+    f32 waterSurface;
+    s32 lightRadius = -1;
+    u8 lightColorY = 0;
+    Player* player;
+    EnArrow* arrow;
+    s32 itemType;
+    u32 toucherFlags;
+    Vec3f tipToFlame;
+    ColliderCylinder* collider1;
+    ColliderCylinder* collider2;
+
+    litTimeScale = torchCount;
+    if(torchCount == 10) {
+        torchCount = 24;
+    }
+    if (func_80042244(globalCtx, &globalCtx->colCtx, this->actor.posRot.pos.x, this->actor.posRot.pos.z, &waterSurface, &dummy) && ((waterSurface - this->actor.posRot.pos.y) > 52.0f )) {
+        this->timer = 0;
+        if (torchType == 1) {
+            Flags_UnsetSwitch(globalCtx, switchFlag);
+            if (torchCount != 0) {
+                this->timer = 1;
+            }
+        }
+    } else {
+        player = PLAYER;
+        itemType = 0;
+        if (this->actor.params & 0x400) {
+            this->timer = -1;
+        }
+        if (torchCount != 0) {
+            if (Flags_GetSwitch(globalCtx, switchFlag)) {
+                if (this->timer == 0) {
+                    this->timer = -1;
+                    if (torchType == 0) {
+                        func_80080480(globalCtx, &this->actor);
+                    }
+                } else if (this->timer > 0) {
+                    this->timer = -1;
+                }
+            } else if (this->timer < 0) {
+                this->timer = 20;
+            }
+        }
+        if (this->collider2.base.acFlags & 2) {
+            toucherFlags = this->collider2.body.acHitItem->toucher.flags;
+            if (toucherFlags & 0x20820) {
+                itemType = 1;
+            }
+        } else if (player->heldItemActionParam == 6) {
+            Math_Vec3f_Diff(&player->swordInfo[0].tip, &this->actor.posRot.pos, &tipToFlame);
+            tipToFlame.y -= 67.0f;
+            if ((SQ(tipToFlame.x) + SQ(tipToFlame.y) + SQ(tipToFlame.z)) < 400.0f) {
+                itemType = -1;
```

Comment:
are you sure about naming this `itemType` since it also depends on what I assume is the location of the deku stick tip
maybe more like `actionType` or `interactType` idk

Reply:
it's -1 if the item interacting is a deku stick and +1 if it's anything else relevant. Maybe interactType is better.